storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/V4FinalizedStateSnapshotStorageLogic.java

```diff
@@ -86,6 +86,25 @@ public void addFinalizedState(
       }
     }
 
+    @Override
+    public void addReconstructedFinalizedState( // TODO repetition
```

Comment:
I doubt you'll be able to avoid the repetition here unfortunately as it needs to access different variables.  So I'd just remove the TODO.